ci/mkpipeline.py

```diff
@@ -390,6 +393,34 @@ def visit(step: PipelineStep) -> None:
     ]
 
 
+def trim_builds(pipeline: Any, coverage: bool) -> None:
+    """Trim unnecessary x86-64/aarch64 builds if all artifacts already exist."""
+
+    def builds_published(arch: Arch) -> bool:
+        repo = mzbuild.Repository(Path("."), arch=arch, coverage=coverage)
+        deps_publish = repo.resolve_dependencies(
+            image for image in repo if image.publish
+        )
+        return deps_publish.check()
+
+    def visit(step: dict[str, Any]) -> None:
+        if step.get("id") == "build-x86_64":
+            if builds_published(Arch.X86_64):
+                step["skip"] = True
+        if step.get("id") == "build-aarch64":
```

Comment:
It is not necessary here to include the command as done in other places (`ident = step.get("id") or step.get("command")`)? If so, when is that necessary? @def-

Reply:
Because the id is build-* in our actual pipelines, we don't have any commands with that name, the command would be something like `bin/ci-builder run stable bin/pyactivate -m ci.test.build`